File: Program2/SharedBool.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class SharedBool
    {
        private Boolean m_bool; // bool value

        /*
         * Shared Bool constructor
         */
         public SharedBool(Boolean boolean)
        {
            m_bool = boolean;
        }

        /*
         * Get bool value
         */
        [MethodImpl(MethodImplOptions.Synchronized)]
        public Boolean Get()
        {
            return m_bool;
        }

        /*
         * Set bool value
         */
        [MethodImpl(MethodImplOptions.Synchronized)]
        public void Set(Boolean boolean)
        {
            m_bool = boolean;
        }
    }
}

File: Program1/SharedBool.java
/*
 * Shared boolean class to allow sharing of boolean between multiple threads
 */
public class SharedBool {
    private boolean m_bool; // bool value

    /*
     * Shared bool constructor
     */
    SharedBool(boolean bool) {
        m_bool = bool;
    }

    /*
     * Synchronized method to get bools value
     */
    synchronized public boolean get() {
        return m_bool;
    }

    /*
     * synchronized method to set bools value
     */
    synchronized public void set(boolean bool) {
        m_bool = bool;
    }
}
